### DISCUSSION TITLE
imageEditor will too large

### DISCUSSION THREAD

**Discussion by 000yuying** [filamentphp/filament#11572] [Help]
### Package

Panel builder

### Package Version

3.2.21

### How can we help you?

When I use 
FileUpload::make('image')
->image()
->imageEditor() 

to crop a photo, the file size becomes larger than the original. 
Has anyone encountered the same issue? How can this problem be resolved?

**Comment by sohailahmad07** [comment]
have you found a solution

**Comment by gaambo** [comment]
### Cause
The root cause seems to be a [known-issue in cropper.js](https://github.com/fengyuanchen/cropperjs?tab=readme-ov-file#known-issues): 
> Known image size increase: When exporting the cropped image on the browser side with the HTMLCanvasElement.toDataURL method, the size of the exported image may be greater than the original image's. This is because the type of the exported image is not the same as the original image. So just pass the original image's type as the first parameter to toDataURL to fix this. For example, if the original type is JPEG, then use cropper.getCroppedCanvas().toDataURL('image/jpeg') to export image.

There's also a [workaround to limit the quality option](https://github.com/fengyuanchen/cropperjs/issues/538#issuecomment-1776279427), but that's only something filament could do in core and currently not possible to configure in your fields: 
> Our solution is to set quality to 0.8, it seems to be a sweet spot for keeping the same size & quality post-edit for most images.

In my case, I could resolve it by just setting `imageResizeTargetWidth` and `imageResizeTargetWidth` to 1024px, because in our application I know, that we will require images only als thumbnails and never larger. That prevents the files from getting too large. 

### Workaround
But before that, I've applied the **following workarounds** (just to get rid of errors) to increase the max file size:
You need to publish the livewire config and change the following (in `config/livewire.php`)
```php
'temporary_file_upload' => [
        // TODO: workaround for images getting larger in file-pond/cropper.js
        'rules' => ['required', 'file', 'max:30000'],
 // [...]
]
```

And on all image fields set:
```php
// workaround for images getting larger in file-pond/cropper.js
// also needs option in config/livewire.php updated
->maxSize(30000)
```

The last step, I'm not sure if I only needed because I use the media-library-manager plugin: 

And in `config/media-library.php` I added
```php
/*
 * The maximum file size of an item in bytes.
 * Adding a larger file will result in an exception.
 */
// 'max_file_size' => 1024 * 1024 * 10, // 10MB
// TODO: workaround for images getting larger in file-pond/cropper.js
'max_file_size' => 1024 * 1024 * 30, // 30MB
```
